OCaml 5.5.0 build failure on Windows

The attempt to upgrade my Windows OCaml installation to 5.5.0 keeps failing, please see the following log. The path `C:\\Users\\me\\AppData\\Local\\opam\\default\\lib\\toplevel:stdlib.cmi` that it tries to access looks odd.
```
$ opam switch create 5.5.0

<><> Installing new switch packages <><><><><><><><><><><><><><><><><><><><>  🐫

Switch invariant: ["ocaml-base-compiler" {= "5.5.0"}]

<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><>  🐫

∗ installed arch-x86_64.1
∗ installed base-bigarray.base
∗ installed base-threads.base
∗ installed base-unix.base
∗ installed compiler-cloning.enabled
∗ installed host-arch-x86_64.1
∗ installed host-system-mingw.1
∗ installed mingw-w64-shims.1.0.1
⬇ retrieved flexdll.0.44  (cached)
∗ installed flexdll.0.44
⬇ retrieved conf-mingw-w64-gcc-x86_64.1  (https://opam.ocaml.org/cache)
∗ installed ocaml-options-vanilla.1
⬇ retrieved ocaml.5.5.0  (https://opam.ocaml.org/cache)
⬇ retrieved ocaml-compiler.5.5.0  (https://opam.ocaml.org/cache)
∗ installed conf-mingw-w64-gcc-x86_64.1
∗ installed ocaml-env-mingw64.1
∗ installed system-mingw.1
[ERROR] The compilation of ocaml-compiler.5.5.0 failed at "sh
        tools/opam/process.sh make -j15
        bba7f261fa998bd3da96a67ddbcea659fa27c48118ab1b7e78414420c73b7b78
        ocaml-compiler enabled --host=x86_64-w64-mingw32
        --prefix=C:\\Users\\me\\AppData\\Local\\opam\\5.5.0
        --docdir=C:\\Users\\me\\AppData\\Local\\opam\\5.5.0\\doc/ocaml
        --with-additional-stublibsdir --with-relative-libdir
        --enable-runtime-search --enable-runtime-search-target=fallback
        --with-flexdll=C:\\Users\\me\\AppData\\Local\\opam\\5.5.0\\share\\flexdll
        --disable-warn-error".

#=== ERROR while compiling ocaml-compiler.5.5.0 ===============================#
# context     2.4.1 | win32/x86_64 |  | https://opam.ocaml.org#169cb087becf658110c92dcaf6e56d94858452d9
# path        C:\Users\me\AppData\Local\opam\5.5.0\.opam-switch\build\ocaml-compiler.5.5.0
# command     C:\Users\me\AppData\Local\opam\.cygwin\root\bin\sh.exe tools/opam/process.sh make -j15 bba7f261fa998bd3da96a67ddbcea659fa27c48118ab1b7e78414420c73b7b78 ocaml-compiler enabled --host=x86_64-w64-mingw32 --prefix=C:\Users\me\AppData\Local\opam\5.5.0 --docdir=C:\Users\me\AppData\Local\opam\5.5.0\doc/ocaml --with-additional-stublibsdir --with-relative-libdir --enable-runtime-search --enable-runtime-search-target=fallback --with-flexdll=C:\Users\me\AppData\Local\opam\5.5.0\share\flexdll --disable-warn-error
# exit-code   2
# env-file    C:\Users\me\AppData\Local\opam\log\ocaml-compiler-5756-9aff61.env
# output-file C:\Users\me\AppData\Local\opam\log\ocaml-compiler-5756-9aff61.out
### output ###
# [...]
# done
# make[1]: Entering directory '/cygdrive/c/Users/me/AppData/Local/opam/5.5.0/.opam-switch/build/ocaml-compiler.5.5.0/otherlibs/runtime_events'
# make[1]: Leaving directory '/cygdrive/c/Users/me/AppData/Local/opam/5.5.0/.opam-switch/build/ocaml-compiler.5.5.0/otherlibs/runtime_events'
# make[1]: Entering directory '/cygdrive/c/Users/me/AppData/Local/opam/5.5.0/.opam-switch/build/ocaml-compiler.5.5.0/otherlibs/unix'
# make[1]: Leaving directory '/cygdrive/c/Users/me/AppData/Local/opam/5.5.0/.opam-switch/build/ocaml-compiler.5.5.0/otherlibs/unix'
# make[1]: Entering directory '/cygdrive/c/Users/me/AppData/Local/opam/5.5.0/.opam-switch/build/ocaml-compiler.5.5.0/otherlibs/str'
# make[1]: Leaving directory '/cygdrive/c/Users/me/AppData/Local/opam/5.5.0/.opam-switch/build/ocaml-compiler.5.5.0/otherlibs/str'
# make[1]: Entering directory '/cygdrive/c/Users/me/AppData/Local/opam/5.5.0/.opam-switch/build/ocaml-compiler.5.5.0/otherlibs/systhreads'
# make[1]: Leaving directory '/cygdrive/c/Users/me/AppData/Local/opam/5.5.0/.opam-switch/build/ocaml-compiler.5.5.0/otherlibs/systhreads'
# ./boot/ocamlrun.exe ./ocaml.exe -nostdlib -I ./stdlib tools/opam/generate.ml clone ocaml-compiler 'ln -sf'
# Fatal error: exception Sys_error("C:\\Users\\me\\AppData\\Local\\opam\\default\\lib\\toplevel:stdlib.cmi: No such file or directory")
# make: *** [Makefile:2782: install] Error 2
```

--- a/Changes
+++ b/Changes
@@ -494,6 +494,9 @@ OCaml 5.5.1
   ````
   (Florian Angeletti, report by Hazem ElMasry, review by Gabriel Scherer)
 
+- #14871, #14914: Ignore OCAMLTOP_INCLUDE_PATH during the build.
+  (David Allsopp, report by Andreas Rossberg, review by Florian Angeletti)
+
 - #14940, fix a memory leak in the OCaml runtime by bounding the size
   of the internal cache of stacks.
   (Vesa Karvonen, Florian Angeletti, review by Gabriel Scherer)

--- a/Makefile.common
+++ b/Makefile.common
@@ -871,3 +871,9 @@ export CYGWIN := $(strip \
 export MSYS := $(strip \
   $(filter-out winsymlinks winsymlinks:%, $(MSYS)) winsymlinks:nativestrict)
 endif
+
+# Invocations of the compilers during the build use -nostdlib which ensures that
+# existing installations and the OCAMLLIB/CAMLLIB environment variables don't
+# affect the build. There isn't an equivalent flag for the toplevel's
+# OCAMLTOP_INCLUDE_PATH, so it's scrubbed from the environment instead.
+unexport OCAMLTOP_INCLUDE_PATH

